### PR TITLE
Hookup update

### PR DESCRIPTION
**Pull request by stevekroh**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
   "homepage": "https://github.com/bikelomatic-complexity/btc-app#readme",
   "dependencies": {
     "blob-util": "^1.2.0",
-    "btc-models": "git+https://github.com/bikelomatic-complexity/btc-models.git#5742417",
+    "btc-models": "^0.2.0",
```
Comment (stevekroh) [inline package.json]: btc-models is on npm now

File: src/js/app.js [hunk 2]
```diff
@@ -41,32 +41,22 @@ import RatingPointCard from './components/point-card/rating-point-card';
...
 import history from './history';
 
-// Builds a redux store given optional middleware.
-// TODO: refactor out this dependency once we only depend on redux-thunk.
-import StoreBuilder from './store';
-
-// Allow direct access to the local PouchDB instance and import the gateway
-// singleton to make working with the database easier.
-import database from './database';
-import gateway from './gateway';
```
Comment (stevekroh) [inline src/js/app.js]: Don't need a special gateway class anymore, we're using backbone-pouch in the btc-models.

File: src/js/app.js [hunk 3]
```diff
@@ -79,18 +69,13 @@ injectTapEventPlugin();
...
-
-  const network = new NetworkManager( store );
-  network.monitor();
+  const network = new NetworkStateAgent( store );
+  network.run();
 
-  const sync = new Sync( database, gateway, store );
-  sync.start();
+  const replicator = new ReplicationAgent( store );
+  replicator.run();
```
Comment (stevekroh) [inline src/js/app.js]: `NetworkStateAgent` and `ReplicationAgent` replace `NetworkManager` and `Sync`. Both are "agents" which subscribe to store updates and react to some change.

File: src/js/app.js [hunk 4]
```diff
@@ -41,32 +41,22 @@ import RatingPointCard from './components/point-card/rating-point-card';
...
+// Import the database to connect it to our models
+import './database';
+import store from './store';
 
 // Sync and NetworkManager allow us to automatically work with the remote
 // PouchDB when the user is online.
-import Sync from './sync';
-import { NetworkManager } from './reducers/network';
+import { NetworkStateAgent } from './reducers/network';
+// import Sync from './sync';
```
Comment (JRJurman) [inline src/js/app.js]: we should remove this if we don't need it...

File: src/js/components/wizard/service-hours.js [hunk 5]
```diff
@@ -1,115 +1,114 @@
...
+
+// The pickers default to 8 AM and 5 PM respectively
+const defaultOpens = new Date( 1970, 0, 1, 8, 0, 0, 0 );
+const defaultClose = new Date( 1970, 0, 1, 17, 0, 0, 0 );
 
 export class ServiceHours extends WizardPage {
   constructor( props ) {
     super( props );
-    bindAll( this, 'onDaySelect', 'addHours', 'removeHours' );
+    bindAll( this, 'addHours', 'removeHours', 'onRowSelection' );
```
Comment (JRJurman) [inline src/js/components/wizard/service-hours.js]: just curious, what does this bindAll do?

**Comment by JRJurman** [comment]
Done reviewing the code, will test on Browser and device, and then if everything is good, I'll give it the ol' +1

**Comment by JRJurman** [comment]
image
Is this supposed to be small? or is there a class that's not be applied here...

**Comment by JRJurman** [comment]
I keep on getting "Replication Error" when starting the app... is that supposed to be happening?

**Comment by JRJurman** [comment]
Trying to create a new service fails on mobile...
image

**Comment by JRJurman** [comment]
The phone number field in add details does not persist...

File: src/js/components/wizard/service-hours.js (hunk 5 above)
Comment (stevekroh) [inline src/js/components/wizard/service-hours.js]: Binds instance methods to this, just like this.addHours = this.addHours.bind( this ). Its good to use when your class has a lot of event handlers.

**Comment by stevekroh** [comment]
1. We can think of different styling options for the location chooser's next button on desktop. Right now it's the way it is for mobile.
2. The backend must be down. That's where the replication errors are coming from.
3. Will look into the other problems.

**Comment by ethanjurman** [comment]
:+1: after resolve

**Pull request merged**